waiter/src/waiter/core.clj

```diff
@@ -837,12 +842,12 @@
                                  (digest/md5 (str service-id (first passwords)))))
    ; This function is only included here for initializing the scheduler above.
    ; Prefer accessing the non-starred version of this function through the routines map.
-   :service-id->service-description-fn* (pc/fnk [[:settings service-description-defaults metric-group-mappings]
-                                                 [:state kv-store]]
+   :service-id->service-description-fn* (pc/fnk [[:settings metric-group-mappings]
```

Comment:
Keeping service-description-defaults as independent from profiles feels doesn't feel awkward, where as sticking service-description-defaults in the profiles map as an empty string key does.

Reply:
Keeping separate also helps reinforce the notion that profiles are not "layered".  We have defaults, and a single profile that can overlay a new set of defaults.

Reply:
Profile providing a different set of layered defaults is how it is represented in the configuration with the different `service-description-defaults` and `profile-config` sections.

The implementation 'optimizes' by pre-computing the profiles to default parameters mapping instead of having to overlay the parameters on every request. It also saves us from having to pass a couple of parameters around every time. I can change the mapping to `nil` or a `:default` keyword to represent the no-explicit profile specified case.

What do you think?

Reply:
They're not equivalent keys in the map.  A profile can be a subset, where as all defaults cannot be a subset.  The set of all default parameters are not a "profile."  The optimization doesn't sound like it saves that much and could be done higher up.  Having separate parameters increases clarity IMO and not worth the "savings" of having separate parameters.

Reply:
Changed the code to pass in two separate arguments: `service-description-defaults` as before and the new `profile->overrides`.